tests/integration/test_feedback.py

```diff
@@ -1,9 +1,13 @@
 """Integration tests for REST API endpoints for providing user feedback."""
 
+import os
+
 import requests
 from fastapi.testclient import TestClient
 
-from ols.app.main import app
+# config file path needs to be set before importing app which tries to load the config
+os.environ["OLS_CONFIG_FILE"] = "tests/config/valid_config.yaml"
+from ols.app.main import app  # noqa: E402
```

Comment:
@syedriko @tisnik @onmete is there a different preferred solution here?

basically our app module loads the config from the OLS_CONFIG_FILE env during import.  We need a way to control what config file is loaded during integration testing where we still want to do a normal init of the application, but we need to control the config file being loaded.  

Probably main.py itself should be refactored so that some of the logic it contains is only executed when run as main and not when imported (i.e. `if __name__ == "__main__"`)

Reply:
it can be done this way. Usually (especially for unit tests) the preferred way is to change environment variables just or one test or for given context via `setup/teardown` mechanism:

```
@patch.dict(os.environ, {"OLS_CONFIG_FILE": "tests/config/valid_config.yaml"})
```

it means that this test won't change behavior of other tests

Reply:
and the import, it's mainly ok for tests

Reply:
Seems alright. You either patch env when you need to provide a full config to test or just patch relevant parts of the actual config object (regardless of what is loaded).

Reply:
thanks.  i'll leave this as is for now then